fireplace/cards/utils.py

```diff
@@ -38,6 +38,50 @@
 AT_MAX_MANA = lambda s: MANA(s) == 10
 
 
+# take a list of weights and a list of card pools and produce a random weighted sample without replacement
+def weighted_card_choice(source, weights, card_sets, count):
```

Comment:
Can we please document the types of the input args (and their sizes, etc. in cases where they would need to match one another), and optionally, use type annotations where appropriate?